Welcome Message

Kankur0 opened the pull request:
Added welcome message when a member joined and allowed the member to request for the personal message to be resent.

Kankur0 committed:
Added welcome message when a member joins to client.js.
Added welcome message resend command via welcome.js controller.

gmoore10 reviewed (commented):
Hi Toby,

Your code looks good. I am curious as to why a Terms of Service welcome action was added instead of a Resend Welcome Message action. According to requirement #3 in the slack channel, it should allow the member to resend the welcome message that was sent when the member joined the server.

I may have overlooked something myself though. Can you confirm?

Kankur0 committed:
Changed command to !resendWelcome to help users understand it’s purpose easier as suggested by Garrett Moore.

Kankur0 commented:
Thanks for the feedback, I've renamed my command and action method to better reflect on it's purpose. Let me know if these changes are more preferred.

Kankur0 committed:
Fixed unhandled promise rejection within welcomeAction.
Simplified welcome message within client.js by removing createDM() as it wasn’t needed and changed sendMessage to just send since sendMessage triggered a deprecation warning .

Kankur0 commented:
Made some slight changes to my code to make it more semantic and removed unneeded code.

gmoore10 reviewed (approved):
Looks good!

reactivepixel commented on bot/controllers/welcome.js:
```diff
@@ -0,0 +1,25 @@
+
+  welcomeAction() {
+    const { message } = this;
+    return `Welcome to the server ${message.member}!\r\nYou can view the terms here!`;
```
Duplicate logic: See ```client.js```

reactivepixel commented on bot/client.js:
```diff
@@ -57,4 +57,17 @@ client.on('message', (message) => {
+    channel.send(`Everybody say hello to ${member.user.username}!`); // Send welcome message
+
+    // Send member a welcome message with terms
+    member.send(`Welcome to the server ${member.user.username}!\r\nYou can view our terms of service here!`);
```
Duplicate logic: See ```welcome.js```

reactivepixel commented on bot/client.js:
```diff
@@ -57,4 +57,17 @@ client.on('message', (message) => {
+client.on('guildMemberAdd', (member) => {
+  // Announce member in general channel
+  const channel = member.guild.channels.find('name', 'general'); // Find & assign channel
+  if (!channel) {
```
Great point of checking for the general channel! Good work in covering this!

reactivepixel commented on bot/client.js:
```diff
@@ -57,4 +57,17 @@ client.on('message', (message) => {
   }
 });
 
+client.on('guildMemberAdd', (member) => {
```
Narrow scoped logic. The framework is built around running multiple controllers against each ```client.on``` while you are singularly doing this. This makes the function inflexible and a one-time use. Should this need extended further it would require a full rewrite of what you have added.

reactivepixel reviewed (commented):
Good work thinking through the logical flow of the potential errors.
